### PR TITLE
Add extra functions & constants from sys/mman.h for Redox

### PR DESCRIPTION
**Pull request by AdminXVII**
This should add support for the mmap crate for Redox.

cc @jackpot51

### CODE REVIEW

**Comment by AdminXVII** [comment]
I don't think the CI failure is due to this PR, as it only changes the redox's definitions.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "libc"
-version = "0.2.70"
+version = "0.2.71"
```
Comment (JohnTitor) [inline Cargo.toml:3]: I'd like to wait for a while to make a release since we recently did.

Comment (AdminXVII) [reply Cargo.toml]: Sure

**Comment by AdminXVII** [comment]
Done

**Pull request merged**